lib/steep/server/worker_process.rb

```diff
@@ -57,6 +57,14 @@ def shutdown
         writer.io.close()
         @wait_thread.join()
       end
+
+      def interrupt
+        return unless @wait_thread.alive?
+        Process.kill "INT", @wait_thread.pid
+        sleep 0.1
+        return unless @wait_thread.alive?
+        Process.kill "TERM", @wait_thread.pid
+      end
```

Comment:
Worker processes can be stopped earlier by sending signals.

In my environment, SIGINT was not effective sometimes. Because of that, I added some lines to send a signal again with SIGTERM.